File: HideBloat/src/main/java/com/aliucord/plugins/hidebloat/patchers/ChannelsInviteButtonPatch.java
package com.aliucord.plugins.hidebloat.patchers;

import android.view.View;

import com.aliucord.plugins.hidebloat.util.Const;
import com.aliucord.plugins.hidebloat.patchers.base.BasePatcher;
import com.aliucord.plugins.hidebloat.util.Util;
import com.discord.widgets.channels.list.WidgetChannelsListAdapter;
import com.discord.widgets.channels.list.items.ChannelListItem;

import top.canyie.pine.Pine;

public class ChannelsInviteButtonPatch extends BasePatcher {

    public ChannelsInviteButtonPatch() throws Exception {
        super(Const.Key.INVITE_BUTTON_CHANNELS_KEY, Const.ViewName.INVITE_BUTTON_CHANNELS_NAME, WidgetChannelsListAdapter.ItemInvite.class.getDeclaredMethod("onConfigure", int.class, ChannelListItem.class));
    }

    @Override
    public void patchBody(Pine.CallFrame callFrame) {
        View itemView = ((WidgetChannelsListAdapter.ItemInvite) callFrame.thisObject).itemView;
        Util.hideViewCompletely(itemView);
        callFrame.setResult(callFrame.getResult());
    }
}

File: plugins/LayoutController/src/main/java/com/aliucord/plugins/layoutcontroller/patchers/ChannelsInviteButtonPatch.kt
package com.aliucord.plugins.layoutcontroller.patchers

import com.aliucord.plugins.layoutcontroller.patchers.base.BasePatcher
import com.aliucord.plugins.layoutcontroller.util.Description
import com.aliucord.plugins.layoutcontroller.util.Key
import com.aliucord.plugins.layoutcontroller.util.hideCompletely
import com.discord.widgets.channels.list.WidgetChannelsListAdapter
import com.discord.widgets.channels.list.items.ChannelListItem
import top.canyie.pine.Pine.CallFrame

class ChannelsInviteButtonPatch : BasePatcher(
    key = Key.INVITE_BUTTON_CHANNELS_KEY,
    description = Description.INVITE_BUTTON_CHANNELS_DESCRIPTION,
    classMember = WidgetChannelsListAdapter.ItemInvite::class.java.getDeclaredMethod(
        "onConfigure",
        Int::class.javaPrimitiveType,
        ChannelListItem::class.java
    )
) {
    override fun patchBody(callFrame: CallFrame) {
        val itemView = (callFrame.thisObject as WidgetChannelsListAdapter.ItemInvite).itemView
        itemView.hideCompletely()
        callFrame.result = callFrame.result
    }
}